tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -1386,27 +1429,38 @@ SQLITE_API int sqlite3_result_zeroblob64(sqlite3_context *, sqlite3_uint64 n) {
 }
 
 // TODO complain
-const void *sqlite3_value_blob(sqlite3_value *) {
-	return nullptr;
+const void *sqlite3_value_blob(sqlite3_value *pVal) {
+	return sqlite3_value_text(pVal);
 }
 
-double sqlite3_value_double(sqlite3_value *) {
-	return 0;
+double sqlite3_value_double(sqlite3_value *pVal) {
+	if (pVal->type == SQLiteTypeValue::FLOAT) {
+		return pVal->u.r;
+	}
+	return 0.0;
 }
 
-int sqlite3_value_int(sqlite3_value *) {
-	return 0;
+int sqlite3_value_int(sqlite3_value *pVal) {
+	return sqlite3_value_int64(pVal);
 }
 
-sqlite3_int64 sqlite3_value_int64(sqlite3_value *) {
+sqlite3_int64 sqlite3_value_int64(sqlite3_value *pVal) {
+	if (pVal->type == SQLiteTypeValue::INTEGER) {
```

Comment:
Same here, I believe we should add a cast here if the underlying value type is different.